libraries/AP_ExternalAHRS/AP_ExternalAHRS_InertialLabs.cpp

```diff
@@ -0,0 +1,659 @@
+            CHECK_SIZE(u.gyro_data_hr);
+            ins_data.gyro = u.gyro_data_hr.tofloat().rfu_to_frd()*DEG_TO_RAD*1.0e-5;
+            break;
+        }
+        case MessageType::BARO_DATA: {
+            CHECK_SIZE(u.baro_data);
+            baro_data.pressure_pa = u.baro_data.pressure_pa2*2;
+            break;
+        }
+        case MessageType::MAG_DATA: {
+            CHECK_SIZE(u.mag_data);
+            mag_data.field = u.mag_data.tofloat().rfu_to_frd()*(10*NTESLA_TO_MGAUSS);
+            break;
+        }
+        case MessageType::ORIENTATION_ANGLES: {
+            CHECK_SIZE(u.orientation_angles);
+            state.quat.from_euler(radians(u.orientation_angles.roll*0.01),
+                                  radians(u.orientation_angles.pitch*0.01),
+                                  radians(u.orientation_angles.yaw*0.01));
+            state.have_quaternion = true;
+            if (last_att_ms == 0) {
+                GCS_SEND_TEXT(MAV_SEVERITY_INFO, "InertialLabs: got link");
+            }
+            last_att_ms = now_ms;
+            break;
+        }
+        case MessageType::VELOCITIES: {
+            CHECK_SIZE(u.velocity);
+            state.velocity = u.velocity.tofloat().rfu_to_frd()*0.01;
+            state.have_velocity = true;
+            last_vel_ms = now_ms;
+            break;
+        }
+        case MessageType::POSITION: {
+            CHECK_SIZE(u.position);
+            state.location.lat = u.position.lat;
+            state.location.lng = u.position.lon;
+            state.location.alt = u.position.alt;
+            state.have_location = true;
+            state.last_location_update_us = AP_HAL::micros();
+            last_pos_ms = now_ms;
+            break;
+        }
+        case MessageType::KF_VEL_COVARIANCE: {
+            CHECK_SIZE(u.kf_vel_covariance);
+            state2.kf_vel_covariance = u.kf_vel_covariance.tofloat() * 0.001;
+            break;
+        }
+        case MessageType::KF_POS_COVARIANCE: {
+            CHECK_SIZE(u.kf_pos_covariance);
+            state2.kf_pos_covariance = u.kf_pos_covariance.tofloat() * 0.001;
+            break;
+        }
+        case MessageType::UNIT_STATUS: {
+            CHECK_SIZE(u.unit_status);
+            state2.unit_status = u.unit_status;
+            break;
+        }
+        case MessageType::GNSS_EXTENDED_INFO: {
+            CHECK_SIZE(u.gnss_extended_info);
+            gps_data.fix_type = u.gnss_extended_info.fix_type+1;
+            state2.gnss_extended_info = u.gnss_extended_info;
+            break;
+        }
+        case MessageType::NUM_SATS: {
+            CHECK_SIZE(u.num_sats);
+            gps_data.satellites_in_view = u.num_sats;
+            break;
+        }
+        case MessageType::GNSS_POSITION: {
+            CHECK_SIZE(u.position);
+            gps_data.latitude = u.position.lat;
+            gps_data.longitude = u.position.lon;
+            gps_data.msl_altitude = u.position.alt;
```

Comment:
Is this WGS-84, or another datum?

Reply:
can be either. Waiting for info from vendor on how to change it using uart API, but I don't think we should hold the PR for that